src/rust/engine/options/src/id.rs

```diff
@@ -1,15 +1,26 @@
 // Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
 // Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+use lazy_static::lazy_static;
 use std::fmt;
 use std::fmt::{Display, Formatter};
 
+use regex::Regex;
+
 #[derive(Clone, Debug, Eq, PartialEq)]
 pub enum Scope {
     Global,
     Scope(String),
 }
 
+lazy_static! {
+    static ref SCOPE_NAME_RE: Regex = Regex::new(r"^[a-z][a-z0-9-]+$").unwrap();
```

Comment:
Just confirming that this intentionally doesn't seem to support any of:

- uppercase
- underscores
- non-ASCII letters

If so, could we add some tests for them?

Reply:
Correct, will add tests.

Reply:
Ah, cool. I've now found:

https://github.com/pantsbuild/pants/blob/39084e99d572af4faf04f19ef922aaeb8f36262a/src/python/pants/option/subsystem.py#L252-L254

https://github.com/pantsbuild/pants/blob/39084e99d572af4faf04f19ef922aaeb8f36262a/src/python/pants/option/subsystem.py#L77

That version of the regex looks slightly different, in particular:

- underscores are valid
- leading digits (and underscores) are supported
- consecutive or trailing `-` isn't supported

So, potentially two changes:

- align this regex to that one, to avoid accidentally breaking users
- as future work (doesn't have to be in this PR), could we potentially centralise the "is valid scope/goal name" code into Rust and have both this Rust code and that Python code call the same functions?

Reply:
Good idea to use the same regex.